server/sonar-web/src/main/js/components/SourceViewer/SourceViewerHeader.tsx

```diff
@@ -17,49 +17,46 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-// @flow
-import React from 'react';
+import * as React from 'react';
 import { Link } from 'react-router';
+import MeasuresOverlay from './components/MeasuresOverlay';
+import { SourceViewerFile } from '../../app/types';
 import QualifierIcon from '../shared/QualifierIcon';
 import FavoriteContainer from '../controls/FavoriteContainer';
-import { getPathUrlAsString, getProjectUrl, getComponentIssuesUrl } from '../../helpers/urls';
+import {
+  getPathUrlAsString,
+  getProjectUrl,
+  getComponentIssuesUrl,
+  getBaseUrl
+} from '../../helpers/urls';
 import { collapsedDirFromPath, fileFromPath } from '../../helpers/path';
 import { translate } from '../../helpers/l10n';
 import { formatMeasure } from '../../helpers/measures';
 
-export default class SourceViewerHeader extends React.PureComponent {
-  /*:: props: {
-    branch?: string,
-    component: {
-      canMarkAsFavorite: boolean,
-      key: string,
-      measures: {
-        coverage?: string,
-        duplicationDensity?: string,
-        issues?: string,
-        lines?: string,
-        tests?: string
-      },
-      path: string,
-      project: string,
-      projectName: string,
-      q: string,
-      subProject?: string,
-      subProjectName?: string,
-      uuid: string
-    },
-    showMeasures: () => void
+interface Props {
+  branch: string | undefined;
+  sourceViewerFile: SourceViewerFile;
+}
+
+interface State {
+  measuresOverlay: boolean;
+}
+
+export default class SourceViewerHeader extends React.PureComponent<Props, State> {
+  state: State = { measuresOverlay: false };
+
+  handleShowMeasuresClick = (event: React.SyntheticEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+    this.setState({ measuresOverlay: true });
   };
-*/
 
-  showMeasures = (e /*: SyntheticInputEvent */) => {
-    e.preventDefault();
-    this.props.showMeasures();
+  handleMeasuresOverlayClose = () => {
+    this.setState({ measuresOverlay: false });
   };
 
-  openInWorkspace = (e /*: SyntheticInputEvent */) => {
-    e.preventDefault();
-    const { key } = this.props.component;
+  openInWorkspace = (event: React.SyntheticEvent<HTMLAnchorElement>) => {
+    event.preventDefault();
+    const { key } = this.props.sourceViewerFile;
     const Workspace = require('../workspace/main').default;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: require() style import is forbidden ([typescript:S3533](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3533))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fmeasures-overlay&issues=AWGu-K4H-_vqYVDUCYpu&open=AWGu-K4H-_vqYVDUCYpu)

[ISSUE_KEY]: AWGu-K4H-_vqYVDUCYpu